src/buf.c

```diff
@@ -93,6 +95,15 @@ void rs_filebuf_free(rs_filebuf_t *fb)
 }
 
 
+void rs_get_filesize(FILE *f, rs_long_t *size)
+{
+    rs_long_t pos = ftell(f);
+    fseek(f, 0, SEEK_END);
```

Comment:
This is more portable, fstat would need another set of different functions depending on the platform. We already have ftell and fseek supposedly supporting large files. Sure, it's unlikely signature would be larger than 4GB, but it doesn't hurt to be on the safe side.

The algorithm falls back to realloc if size is not available, so it should work for pipes...